### PR TITLE
Feature: database integrate w/ Mongoose

### PR DESCRIPTION
**Pull request by bryantee**
Needed before merge:

- [x] Setup proper status code returns
- [x] Setup error handlers
- [x] Code cleanup
- [x] Remove `old-server.js` from repo (unused, only there for reference)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -6,9 +6,41 @@ const config = require('./config');
 
 const app = express();
 
-app.use((bodyParser.json());
+app.use((bodyParser.json()));
```
Comment (amcavinue) [inline server.js:10]: Good catch!

File: server.js [hunk 2]
```diff
@@ -35,6 +35,31 @@ app.post('/items', function(req, res) {
   });
 });
 
+app.put('/items/:id', function(req, res) {
+  let id = req.params.id;
+  let newName = req.body.name;
+
+  if (id === 'undefined') res.status(400).send('Bad request, invalid ID');
+  if (newName === 'undefined' || newName === null) res.status(400).send('Bad request, no name given'); //This isn't working yet
```
Comment (amcavinue) [inline server.js]: My mentor introduced me to the idea of "clean commits". What this means is if you have code that throws errors, breaks other code, doesn't belong, etc., to fix/remove it before committing. That way you can roll back to the commit in the future, without unintentionally breaking things. 

So if this line does any of that, you may want to add a TODO and just comment it out -- my suggestion at least.

**Comment by amcavinue** [comment]
Code looks good. Very clean, readable, and makes logical sense. Not too many comments which doesn't bother me but other people might like them.

File: server.js (hunk 2 above)
Comment (bryantee) [reply server.js]: ^ I like this philosophy @amcavinue. I was getting a little lazy last night at commit :sleeping:

**Pull request merged**